src/nitsm/tsmcontext.py

```diff
@@ -255,11 +265,21 @@ def set_global_data(self, data_id: str, data: _Any) -> None:
 
         Args:
             data_id: A unique ID to distinguish the data.
-            data: A data item to store and later retrieve using the specified data_id . If the data
+            data: A data item to store and later retrieve using the specified data_id. If the data
                 item is None, the method deletes the data with the specified data_id if it exists.
         """
 
-        return self._context.SetGlobalData(data_id, data)
+        if data is not None:
+            py_data_id = id(data)
+            none_ = self._context.SetGlobalData(data_id, py_data_id)
```

Comment:
The only benefit I can think of for using the TSM global data API here is to make the data available to non-python code modules. But that won't work here anyway because we're storing a language-specific ID that is not usable in other languages. Similarly, the get_global_data assumes the stored data is python data.
I think we need to choose one of the following options:
1. Avoid using TSM API altogether and document that global data cannot be used across languages.
2. Use the TSM API, but support basic types (and arrays of basic types) that could be stored by code modules written in different languages.

My preference is option 1 because I prefer to keep things simple until there is a need to do otherwise. I don't think there is an immediate need for cross-language support for global data, but that's something you might want to check.